lua/plenary/job.lua

```diff
@@ -370,13 +370,18 @@ function Job:_execute()
     if Job.is_job(self.writer) then
       self.writer:_execute()
     elseif type(self.writer) == 'table' and vim.tbl_islist(self.writer) then
-      for _, v in ipairs(self.writer) do
-        self.stdin:write(v .. '\n')
+      for i, v in ipairs(self.writer) do
+        self.stdin:write(v)
+        self.stdin:write('\n', function()
+          if i == #self.writer then
+            self.stdin:close()
+          end
+        end)
```

Comment:
do the if check outside of the loop. Only schedule the function if it is the last one.

Also, `local writer_len = #self.writer` so that we don't look it up each time.

Reply:
What about if we just write an empty string after the loop and close in the callback there?
```lua
      for i, v in ipairs(self.writer) do
        self.stdin:write(v)
        self.stdin:write('\n')
      end
      self.stdin:write('', function()
        self.stdin:close()
      end)
```
Does one extra syscall but maybe that's better than N comparisons?